### ISSUE TITLE
New Rule: Disallow the use of spread (`...`) of `reduce` accumulators

### ISSUE THREAD

**Issue by guilhermesimoes** [eslint/eslint#17859]
### Rule details

Disallow the use of spread (`...`) of `reduce` accumulators

### Related ECMAScript feature

spread

### What type of rule is this?

Warns about a potential problem

### Example code

```js
items.reduce((acc, item) => ({ ...acc, [item.id]: item }), {})
```
```js
items.reduce(function (acc, item) {
  return { ...acc, [item.id]: item };
}, {})
```
```js
itemsArrays.reduce((acc, itemsArray) => ([ ...acc, ...itemsArray ]), [])
```

### Why should this rule be in the core instead of a plugin?

Spread syntax should be avoided on accumulators (like those in `reduce`) because it causes a time complexity of O(n^2) instead of O(n).

https://www.richsnapp.com/article/2019/06-09-reduce-spread-anti-pattern
https://prateeksurana.me/blog/why-using-object-spread-with-reduce-bad-idea/

This rule is also present in [biome](https://biomejs.dev/linter/rules/no-accumulating-spread/) and [oxlint](https://github.com/oxc-project/oxc/pull/1607)

### Participation

- [X] I am willing to submit a pull request to implement this rule.

**Comment by bmish** [comment]
It looks like eslint-plugin-unicorn may have some rules to cover this:
* https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1335

**Comment by nzakas** [comment]
Sorry, this is a bit too specific to be a core rule. You can always create a custom rule or use the one that @bmish mentioned above.
